lib/Doctrine/ORM/Query/ParserResult.php

```diff
@@ -122,4 +130,20 @@ public function getSqlParameterPositions($dqlPosition)
     {
         return $this->parameterMappings[$dqlPosition];
     }
+
+    public function __wakeup(): void
+    {
+        $this->__unserialize((array) $this);
+    }
+
+    /** @param array<string, mixed> $data */
+    public function __unserialize(array $data): void
+    {
+        foreach (self::LEGACY_PROPERTY_MAPPING as $property => $legacyProperty) {
+            $this->$property = $data[sprintf("\0%s\0%s", self::class, $legacyProperty)]
+                ?? $data[sprintf("\0%s\0%s", self::class, $property)]
```

Comment:
I would lookup this key first, as it's the happy path.

Reply:
That won't work, unfortunately. When `__wakeup()` is called (which is the case for PHP < 7.4) with a payload created with 2.14, both keys are present.